paimon-core/src/main/java/org/apache/paimon/operation/AbstractFileStoreScan.java

```diff
@@ -237,19 +249,26 @@ private Pair<Snapshot, List<ManifestEntry>> doPlan(
             }
         }
 
+        long startDataFiles =
+                manifests.stream().mapToLong(f -> f.numAddedFiles() + f.numDeletedFiles()).sum();
+
+        AtomicLong cntEntries = new AtomicLong(0);
         Iterable<ManifestEntry> entries =
                 ParallellyExecuteUtils.parallelismBatchIterable(
                         files ->
                                 files.parallelStream()
                                         .filter(this::filterManifestFileMeta)
                                         .flatMap(m -> readManifest.apply(m).stream())
                                         .filter(this::filterByStats)
+                                        .peek(e -> cntEntries.getAndIncrement())
```

Comment:
Why not directly increase `cntEntries` by the size of the collected list? If you use `peek`, `cntEntries` will be increased by every element in this list. Note that this stream is parallelized, so there will be many concurrent writes which affect the performance.